### ISSUE TITLE
Issues about the cross_validation function

### ISSUE THREAD

**Issue by Layonhuuu** [Nixtla/neuralforecast#853]
### What happened + What you expected to happen

The bug

When using the cross_validation function, assign val_size and test_size respectively. Two different situations occurred, as follows:
When the target value corresponding to test_size in the data set is assigned to 0, the prediction results of Y_hat_df are very poor.
When retaining the target value corresponding to test_size in the data set, the prediction results of Y_hat_df are good.
Is there any information leakage in the cross_validation function?
Because for the test_size part of the data set, the target value of test_size should not be known when training the model.
[IMF2-part.csv](https://github.com/Nixtla/neuralforecast/files/13858758/IMF2-part.csv)
[IMF2-part_test_Y=0.csv](https://github.com/Nixtla/neuralforecast/files/13858759/IMF2-part_test_Y.0.csv)

### Versions / Dependencies

neuralforecast 1.6.4（cloned and installed from github）
Python Version = 3.9.18

### Reproduction script

```
import optuna
import pandas as pd
import warnings
warnings.filterwarnings("ignore")
import matplotlib.pyplot as plt
from neuralforecast.core import NeuralForecast
from neuralforecast.auto import AutoDeepAR
import optuna.storages

data_test_original = pd.read_csv("IMF2-part.csv")
data_test_original['ds'] = pd.to_datetime(data_test_original['ds'], unit='s')
val_size = 64*320
test_size = 64*320
horizon =1  
models_test_original = [AutoDeepAR(h=horizon,
                      num_samples=10,
                      backend='optuna',
                      search_alg=optuna.samplers.NSGAIISampler(seed = 1),
)
]
nf_test_original = NeuralForecast(models=models_test_original, freq='s')
Y_hat_df_test_original = nf_test_original.cross_validation(df=data_test_original, val_size=val_size,
                                test_size=test_size, n_windows=None)

plt.plot(Y_hat_df_test_original['ds'], Y_hat_df_test_original['AutoDeepAR-median'])
plt.plot(Y_hat_df_test_original['ds'], Y_hat_df_test_original['y'])
plt.xlabel('Date')
plt.ylabel('IMF2')
plt.title("test_original")
plt.grid()

data_test_Y0 = pd.read_csv("IMF2-part_test_Y=0.csv")
data_test_Y0['ds'] = pd.to_datetime(data_test_Y0['ds'], unit='s')
val_size = 64*320
test_size = 64*320
horizon =1  
models_test_Y0 = [AutoDeepAR(h=horizon,
                      num_samples=10,
                      backend='optuna',
                      search_alg=optuna.samplers.NSGAIISampler(seed = 1),
)
]
nf_test_Y0 = NeuralForecast(models=models_test_Y0, freq='s')
Y_hat_df_test_Y0 = nf_test_Y0.cross_validation(df=data_test_Y0, val_size=val_size,
                                test_size=test_size, n_windows=None)

plt.plot(Y_hat_df_test_Y0['ds'], Y_hat_df_test_Y0['AutoDeepAR-median'])
plt.xlabel('Date')
plt.ylabel('IMF2')
plt.title("test_Y=0")
plt.grid()
```
```

class AutoDeepAR(BaseAuto):
    default_config = {
            "input_size_multiplier": [8,16,24,32,40,48,56,64],
            "h": None,
            "lstm_hidden_size": tune.choice([8,16,32,64,80,96,112,128]),
            "lstm_n_layers": tune.randint(1, 5),
            "lstm_dropout": tune.uniform(0.1, 0.5),
            "learning_rate": tune.loguniform(1e-5, 1e-1),
            "scaler_type": tune.choice(["robust", "minmax1", "identity", "standard"]),
            "max_steps": tune.choice([200,400,600,800,1000]),
            "batch_size": tune.choice([4,8,12,16,20,24,32,40,48]),
            "windows_batch_size":tune.choice([4,8,12,16,20,24,32,40,48]),
            "futr_exog_list": [  
                                'railway', 
                                'SPEED',
                              ],
           "num_lr_decays": 50,
           "random_seed": tune.randint(1, 20),

        }
```

```
        loss=DistributionLoss(
            distribution="StudentT", level=[80, 90], return_params=False
        ),
        valid_loss=MQLoss(level=[80, 90]),
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by cchallu** [comment]
Hi @Layonhuuu. The `cross_validation` function does use the values of the test set; this is not a leakage. The predictions of the test set are rolled, predicting one window of size `h` at a time, moving `step_size` timestamps. Each time it moves the window, it uses the latest true values as inputs. This is the standard cross-validation approach in time series. See image below for an illustration of this process.

In your case, the issue is that the horizon is 1, so the model forecasts only one value at a time using the latest information. To predict the entire test set at once then horizon should equal the test size.

<img width="755" alt="image" src="https://github.com/Nixtla/neuralforecast/assets/31133398/603c9236-897f-4801-9100-8b5293367ee4">

**Comment by Layonhuuu** [comment]
Thanks a lot！
